### PR TITLE
Improve socket connection error logging in `SyftWebSocketDataSource`

### PR DESCRIPTION
**Pull request by karlhigley**
This shows both the exception and the cause in the logs, which makes
it easier to troubleshoot errors.

### CODE REVIEW

**Comment by karlhigley** [comment]
@mccorby Just a small tweak to show more info in the logs, which helped me when troubleshooting network issues. It's possible for the exception to be caused by another exception, so there can be multiple causes; I couldn't figure out a quick way to show all the causes, since at some point the cause of the exception is itself. This seems a like a small step in the right direction though.

File: app/src/main/java/com/mccorby/openmined/worker/datasource/SyftWebSocketDataSource.kt [hunk 1]
```diff
@@ -44,8 +44,11 @@ class SyftWebSocketDataSource(private val webSocketUrl: String, private val clie
         socket.connect()
     }
 
-    private fun logError(args: Array<Any>) {
-        args.forEach { print(" $it") }
+    private fun logError(args: Array<Any>): String {
```
Comment (mccorby) [inline app/src/main/java/com/mccorby/openmined/worker/datasource/SyftWebSocketDataSource.kt]:
```suggestion
    private fun logError(args: Array<Any>): String {
        return args.joinToString("\n") {
            (it as Exception).run {
                "$it, with cause: ${it.cause}"
            }
        }
    }
```

File: app/src/main/java/com/mccorby/openmined/worker/datasource/SyftWebSocketDataSource.kt [hunk 2]
```diff
@@ -44,8 +44,17 @@ class SyftWebSocketDataSource(private val webSocketUrl: String, private val clie
...
+        return args.joinToString("\n") {
+            (it as Exception).run {
+                "$it, with cause: ${it.cause}"
+            }
+        }
+    }
+        return args.map {
```
Comment (mccorby) [inline app/src/main/java/com/mccorby/openmined/worker/datasource/SyftWebSocketDataSource.kt]: You forgot to remove this :)

Comment (karlhigley) [reply app/src/main/java/com/mccorby/openmined/worker/datasource/SyftWebSocketDataSource.kt]: And here I thought the GH feature that allowed me to directly commit your suggestion would do my job for me. Oops!

**Pull request merged**